compiler/rustc_mir_transform/src/elaborate_drops.rs

```diff
@@ -38,25 +38,24 @@ impl<'tcx> MirPass<'tcx> for ElaborateDrops {
         };
         let un_derefer = UnDerefer { tcx: tcx, derefer_sidetable: side_table };
         let elaborate_patch = {
-            let body = &*body;
             let env = MoveDataParamEnv { move_data, param_env };
-            let dead_unwinds = find_dead_unwinds(tcx, body, &env, &un_derefer);
+            remove_dead_unwinds(tcx, body, &env, &un_derefer);
 
             let inits = MaybeInitializedPlaces::new(tcx, body, &env)
                 .into_engine(tcx, body)
-                .dead_unwinds(&dead_unwinds)
                 .pass_name("elaborate_drops")
                 .iterate_to_fixpoint()
                 .into_results_cursor(body);
 
             let uninits = MaybeUninitializedPlaces::new(tcx, body, &env)
                 .mark_inactive_variants_as_uninit()
                 .into_engine(tcx, body)
-                .dead_unwinds(&dead_unwinds)
                 .pass_name("elaborate_drops")
                 .iterate_to_fixpoint()
                 .into_results_cursor(body);
 
+            let reachable = traversal::reachable_as_bitset(body);
```

Comment:
would it be possible to remove unreachable blocks here (or have something like `remove_dead_unwinds` as a separate pass that also prune blocks) so that we don't have to check for dead blocks when processing later?

Reply:
Yeah, it probably would be more elegant to remove dead blocks, instead of adding those extra reachability checks. The motivation behind that approach was to reuse move data between part that identifies dead unwinds and drop elaboration proper. But, maybe this isn't performance sensitive - will have to verify that.

Reply:
Due to [the cost of an additional execution MoveData::gather_moves](https://github.com/rust-lang/rust/pull/106430#issuecomment-1371749609), I reverted back to the original approach.